src/experiments.ts

```diff
@@ -0,0 +1,182 @@
+  public?: boolean;
+  docsUri?: string;
+  default?: boolean;
+}
+
+function experiment<Name extends string>(
+  name: Name,
+  shortDescription: string,
+  opts?: {
+    fullDescription?: string;
+    public?: boolean;
+    docsUri?: string;
+    default?: boolean;
+  }
+): Experiment<Name> {
+  return {
+    ...opts,
+    name,
+    shortDescription,
+  } as Experiment<Name>;
+}
+
+export const EXPERIMENTS = Object.freeze([
+  // meta:
+  experiment("experiments", "enables the experiments family of commands"),
+
+  // Realtime Database experiments
+  experiment("rtdbrules", "TODO"),
+  experiment("rtdbmanagement", "TODO"),
+
+  // Extensions experiments
+  experiment("ext", "Enables the 'ext:sources:create' command"),
+  experiment("extdev", "Enalbes the 'ext:dev' family of commands"),
+
+  // Cloud Functions for Firebase experiments
+  experiment("pythonfunctions", "Python support for Cloud Functions for Firebase", {
+    fullDescription:
+      "Adds the ability to initializea and deploy Cloud " +
+      "Functions for Firebase in Python. While this feature is experimental " +
+      "breaking API changes are allowed in MINOR API revisions",
+  }),
+  experiment("golang", "Golang support for Cloud Functions for Firebase. Does not work.", {
+    fullDescription:
+      "Code paths to play with adding support for Golang to Cloud " +
+      "Functions for Firebase. This was written to target 1st gen functions, which " +
+      "will never be released (all new languages will be 2nd gen only). If you " +
+      "are lurking in our experiments folder and are excited about the prospect of " +
+      "Golang support, tell our support team.",
+  }),
+  experiment(
+    "deletegcfartifacts",
+    `Add the ${clc.bold(
+      "firebase functions:deletegcfartifacts"
+    )} command to purge docker build images`,
+    {
+      fullDescription:
+        `Add the ${clc.bold("firebase functions:deletegcfartifacts")}` +
+        "command. Google Cloud Functions creates Docker images when building your " +
+        "functions. Cloud Functions for Firebase automatically cleans up these " +
+        "images for you on deploy. Customers who predated this cleanup, or customers " +
+        "who also deploy Google Cloud Functions with non-Firebase tooling may have " +
+        "old Docker images stored in either Google Container Repository or Artifact " +
+        `Registry. The ${clc.bold("firebase functions:deletegcfartifacts")} command ` +
+        "will delete all Docker images created by Google Cloud Functions irrespective " +
+        "of how that image was created.",
+      public: true,
+    }
+  ),
+  experiment("functionsparams", "Adds support for paramaterizing functions deployments"),
+  experiment(
+    "skipdeployingnoopfunctions",
+    "Detect that there have been no changes to a function and skip deployment"
+  ),
+
+  // Emulator experiments
+  experiment("emulatoruisnapshot", "Load pre-release versions of hte emulator UI"),
```

Comment:
@yuchenshi (Feel free to delegate to someone who now owns this). Do you want to have a different short description? A full description? Should this be an experiment that customers can see when they call `firebase experiments:list`?